src/main/scala/org/constellation/genesis/Genesis.scala

```diff
@@ -23,156 +25,144 @@ object Genesis extends StrictLogging {
-          TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
-        )
-      )(CoinbaseKey)
-    }
-  }
+    genesisSOE: SignedObservationEdge,
+    coinbaseKey: KeyPair
+  )(dao: DAO): F[CheckpointBlock] =
+    for {
+      dummyTransactionSrc <- Sync[F].delay(KeyUtils.makeKeyPair().getPublic.toId.address)
+      dummyTransactionDst <- Sync[F].delay(KeyUtils.makeKeyPair().getPublic.toId.address)
+      dummyTx <- LiftIO[F].liftIO {
+        dao.transactionService.createDummyTransaction(dummyTransactionSrc, dummyTransactionDst, coinbaseKey)
+      }
+      cb <- Sync[F].delay {
+        CheckpointBlock.createCheckpointBlock(
+          Seq(dummyTx),
+          Seq(
+            TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
+            TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
+          )
+        )(coinbaseKey)
+      }
+    } yield cb
 
   // TODO: Make F[Unit]
-  def createGenesisObservation(
+  def createGenesisObservation[F[_]: Sync: LiftIO](
     allocAccountBalances: Seq[AccountBalance] = Seq.empty
-  )(implicit dao: DAO): GenesisObservation = {
+  )(dao: DAO): F[GenesisObservation] =
     for {
-      genesis <- createDistributionTransactions[IO](allocAccountBalances)
-        .map(createGenesisBlock)
-      emptyA <- createEmptyBlockFromGenesis[IO](genesis.soe)
-      emptyB <- createEmptyBlockFromGenesis[IO](genesis.soe)
+      // TODO: That key should be provided externally by the genesis creator.
+      // We can use Node's KeyPair as well.
+      coinbaseKey <- Sync[F].delay(KeyUtils.makeKeyPair())
+      genesis <- createDistributionTransactions[F](allocAccountBalances, coinbaseKey)(dao)
+        .map(createGenesisBlock(_, coinbaseKey))
+      emptyA <- createEmptyBlockFromGenesis[F](genesis.soe, coinbaseKey)(dao)
+      emptyB <- createEmptyBlockFromGenesis[F](genesis.soe, coinbaseKey)(dao)
     } yield GenesisObservation(genesis, emptyA, emptyB)
-  }.unsafeRunSync() // TODO: Get rid of unsafeRunSync and fix all the unit tests
-
-  def start()(implicit dao: DAO): Unit = {
-    val startIO = for {
-      _ <- dao.cluster.setParticipatedInGenesisFlow(true)
-      _ <- dao.cluster.setParticipatedInRollbackFlow(false)
-      _ <- dao.cluster.setJoinedAsInitialFacilitator(true)
-      _ <- dao.cluster.setOwnJoinedHeight(0L)
-      genesisObservation = createGenesisObservation(dao.nodeConfig.allocAccountBalances)
-      _ <- IO.delay {
-        acceptGenesis(genesisObservation)
-      }
-    } yield genesisObservation
 
-    startIO.unsafeRunSync()
-  }
+  def start[F[_]: Sync: LiftIO](dao: DAO): F[Unit] =
+    for {
+      _ <- LiftIO[F].liftIO(dao.cluster.setParticipatedInGenesisFlow(true))
+      _ <- LiftIO[F].liftIO(dao.cluster.setParticipatedInRollbackFlow(false))
+      _ <- LiftIO[F].liftIO(dao.cluster.setJoinedAsInitialFacilitator(true))
+      _ <- LiftIO[F].liftIO(dao.cluster.setOwnJoinedHeight(0L))
```

Comment:
I believe you should pass a cluster `Cluster[F]` dependency there. The same with others `dao` usages. otherwise, you can receive a null pointer exception during the runtime if there is a potentially circular referency